### ISSUE TITLE
[a11y]: Many violations when there's more than one table with expansions

### ISSUE THREAD

**Issue by jinhua-ibm** [carbon-design-system/carbon#19366]
### Package

@carbon/react

### Package version

1.76.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

While having 2 tables with `<TableExpandRow>` and `<TableExpandHeader>`, it will have lots of violations 

<img width="2539" alt="Image" src="https://github.com/user-attachments/assets/4b3e5978-4554-4275-88c7-bf41cf780464" />

### Reproduction/example

https://stackblitz.com/edit/github-u7nfga-b9nzyzrg?file=src%2FApp.jsx,package.json

### Steps to reproduce

Create 2 tables, with <TableExpandHeader> and <TableExpandRow> and perform a scan.
Also, the placement could affect it as well, the demo is showing the case where they are showing on the z-index. But, there could be an issue too when z-index is different, one covers another.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Rudra1402** [comment]
Hello @preetibansalui, 
The screenshot attached to the issue shows multiple duplicate ID problems, such as `The <tr> element has the id "expanded-row-a" that is already in use`. I would like to contribute towards fixing them. I have tested the changes locally and am just waiting for approval to make a partial contribution to this issue.

**Comment by preetibansalui** [comment]
Hi @Rudra1402, Thanks for looking into it! Please go ahead and raise a PR.

**Comment by tay1orjones** [comment]
Fixed via https://github.com/carbon-design-system/carbon/pull/20633
